### PR TITLE
Fix child search, improve download, add create_folder helper

### PR DESCRIPTION
**Pull request by xsawyerx**

### CODE REVIEW

**Review by atoomic — APPROVED** [review]
LGTM one question for the `parents` sanity part

File: lib/Net/Google/Drive/Simple/V3.pm [hunk 1]
```diff
@@ -593,6 +593,18 @@ sub copy_file {
     return $self->_handle_api_method( $info, $options );
 }
 
+# Helper
+###########################################
+sub create_folder {
+###########################################
+    my ( $self, $name, $parent_id ) = @_;
```
[the remark this answers is not in the document]
Comment (xsawyerx) [reply lib/Net/Google/Drive/Simple/V3.pm:600]: It isn't optional. We need to know where to put the new folder.

File: lib/Net/Google/Drive/Simple/V3.pm [hunk 2]
```diff
@@ -2442,6 +2467,19 @@ This is also known as C<files.copy>.
...
+
+    my $folder_data = $gd->create_folder( $name, $parent_id );
+    print $folder_data->{'id'};
+
+This method is just for convenience. It's effectively:
+
+        $self->create_file({
+            'name'     => $name,
+            'mimeType' => "application/vnd.google-apps.folder",
+            'parents'  => [$parent_id],
```
[the remark this answers is not in the document]
Comment (xsawyerx) [reply lib/Net/Google/Drive/Simple/V3.pm:2480]: In theory, yes, but I couldn't figure out the cases when that happens.

File: lib/Net/Google/Drive/Simple/V3.pm (hunk 1 above)
[the remark this answers is not in the document]
Comment (xsawyerx) [reply lib/Net/Google/Drive/Simple/V3.pm:600]: Added the checks.

**Review by atoomic — APPROVED** [review]

**Pull request merged**